lib/observer/src/etop_txt.erl

```diff
@@ -69,21 +80,120 @@ do_update(Fd,Info,Prev,Config) ->
 		       RQ,Atom,Ets])
     end,
     io:nl(Fd),
-    writepinfo_header(Fd),
-    writesingleline(Fd),
-    writepinfo(Fd,Info#etop_info.procinfo,modifier(Fd)),
-    writedoubleline(Fd),
+    writepinfo_header(Fd, FieldWidths),
+    writesingleline(Fd, FieldWidths),
+    writepinfo(Fd, Info#etop_info.procinfo, modifier(Fd), FieldWidths),
+    writedoubleline(Fd, FieldWidths),
     io:nl(Fd),
     Info.
 
-writepinfo_header(Fd) ->
-    io:fwrite(Fd,"Pid            Name or Initial Func    Time    Reds  Memory    MsgQ Current Function~n",[]).
 
-writesingleline(Fd) ->
-    io:fwrite(Fd,"----------------------------------------------------------------------------------------~n",[]).
-writedoubleline(Fd) ->
-    io:fwrite(Fd,"========================================================================================~n",[]).
- 
+calc_field_widths(ProcInfoL) ->
+    Cols = case io:columns() of
+               {ok, IoCols}     -> IoCols;
```

Comment:
It is actually implicit because get_width/3 doesn't return
a width larger than `8` if the column width budget is
depleted, i.e. `ColsLeft - MaxWidth > 0`.

It doesn't change the etop behavior having
`Cols < ?DEFAULT_WIDTH`, however the intent might be
more clear.

FWIW I changed it for readability and future proofing.

I also commented how get_width/3 works.